python/lsst/pipe/base/activator.py

```diff
@@ -0,0 +1,449 @@
+        log.info("Enabling cProfile profiling")
+    profile.enable()
+    yield profile
+    profile.disable()
+    profile.dump_stats(filename)
+    if log is not None:
+        log.info("cProfile stats written to %s" % filename)
+
+
+class TaskNameError(Exception):
+    """
+    Exception Class to handle conflict with task and config names,
+    might be replaced for a more general one, or even RunTimeError
+    """
+
+    def __init__(self, msg, errs):
+        """
+        Initiate the class, gets the errs and message
+        """
+        super(TaskNameError, self).__init__(msg)
+        self.errs = errs
+
+
+class Activator(object):
+    """
+    Hook for other activators
+    """
+
+    def __init__(self):
+        """
+        This might grow into a common initialization, right now
+        is overwritten
+        """
+        pass
+
+    def activate(self):
+        """
+        This is the required method to execute the activator
+        """
+        pass
+
+
+class CmdLineActivator(Activator):
+    """
+    Activator for the command line: This Class takes the higher level SuperTask as input and
+    executes in the command line, similarly to CmdLineTask but this is a generic way to run (
+    Super)Task. It takes the name of the task from the command line along all the arguments to
+    run the task.
+    Look http://dmtn-002.lsst.io to see examples on how to invoke the activator
+    """
+
+    def __init__(self, SuperTask, parsed_cmd, return_results=False):
+        """
+        Constructs the CmdLineActivator for the SuperTask Class parsed, given the command line
+        options
+
+        :param SuperTask: The actual SuperTask Class that the activator will execute activate
+        :param parsed_cmd: The parsed arguments in the command line, relevant to the task,
+        not to the activator
+        :param return_results: Whether or not return the results, similar to CmdLineTask
+        """
+        super(CmdLineActivator, self).__init__()
+
+        self.SuperTask = SuperTask
+        self.return_results = bool(return_results)
+        if parsed_cmd:
+            self.config = parsed_cmd.config
+            self.log = parsed_cmd.log
+            self.do_raise = bool(parsed_cmd.doraise)
+            self.clobber_config = bool(parsed_cmd.clobberConfig)
+            # self.do_backup = not bool(parsed_cmd.noBackupConfig)
+            self.parsed_cmd = parsed_cmd
+        self.num_processes = 1  # int(getattr(parsed_cmd, 'processes', 1)), lets fix it for now
+
+
+    def make_task(self):
+        """
+        It creates another instance of the SuperTask passed to the activator, it returns the
+        SuperTask instance. Similar to MakeTask in CmdlineTask
+        """
+        return self.SuperTask.__class__(config=self.config, log=self.log, activator='cmdLine')
+
+    def precall(self):
+        """
+        A hook similar to the CmdLineTask one, is True by default and need to copy it from it,
+        which is TBD.
+
```

Comment:
I had issues when I copied some of these methods  from CmdLineTask.py, so I just left this as a hook as I need to bring more methods back to the activator. I'll add a note on this as it has not been fully imported